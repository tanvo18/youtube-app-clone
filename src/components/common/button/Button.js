import React from 'react';
import ButtonStyled from './ButtonStyled';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { 
    btnClass,
    text,
    btnBackground,
    textColor,
    handleClick,
    btnFontSize,
    borderRadius,
    borderColor,
    hoverBgColor 
  } = props;
  
  return (
    <ButtonStyled
      className={btnClass}
      onClick={handleClick}
      textColor={textColor}
      btnBackground={btnBackground}
      fontSize={btnFontSize}
      borderRadius={borderRadius}
      borderColor={borderColor}
      hoverBgColor={hoverBgColor}
    >
      {text}
    </ButtonStyled>
  );
};

Button.defaultProps = {
  textColor: '#fff',
  btnBackground: '#fff',
  btnFontSize: '16'
};

Button.propsType = {
  className: PropTypes.string,
  textColor: PropTypes.string,
  btnBackground: PropTypes.string.isRequired,
  fontSize: PropTypes.string.isRequired,
  borderColor: PropTypes.string.isRequired,
  hoverBgColor: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

export default Button;