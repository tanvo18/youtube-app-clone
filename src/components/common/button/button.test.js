import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

const props = {
  btnClass: 'btn-test', 
  text: 'Button test',
  btnBackground: 'transparent',
  textColor: '#8f8f8f',
  btnFontSize: '16px',
  borderRadius: '4px',
  borderColor: '#000',
  hoverBgColor: '#d9d9d9'
};

describe('test button', () => {
  it('should render a button', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Button handleClick={handleClick} {...props} />);
    expect(wrapper.props().children).toBe('Button test');
  });

  it('should call function handle click', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Button handleClick={handleClick} {...props} />);
    wrapper.simulate('click');
    expect(handleClick.mock.calls.length).toBe(1);
  });

  it('should render correctly', () => {
    const handleClick = jest.fn();
    const tree = renderer
      .create(<Button handleClick={handleClick} {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});