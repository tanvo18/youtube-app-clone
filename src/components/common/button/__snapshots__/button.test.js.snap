// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button should render correctly 1`] = `
<button
  className="btn-test sc-bdVaJa gVmgxe"
  fontSize="16px"
  onClick={[Function]}
>
  Button test
</button>
`;
